python-ecosys/debugpy/debugpy/server/pdb_adapter.py

```diff
@@ -0,0 +1,285 @@
+        self.continue_event = False
+        
+        # Process DAP messages while waiting for continue
+        self._debug_print("[PDB] Waiting for continue command...")
+        while not self.continue_event:
+            # Process any pending DAP messages (scopes, variables, etc.)
+            if hasattr(self, '_debug_session'):
+                self._debug_session.process_pending_messages()
+            time.sleep(0.01)
+            
+    def get_stack_trace(self):
+        """Get the current stack trace."""
+        if not self.current_frame:
+            return []
+            
+        frames = []
+        frame = self.current_frame
+        frame_id = 0
+        
+        while frame:
+            filename = frame.f_code.co_filename
+            name = frame.f_code.co_name
+            line = frame.f_lineno
+            
+            # Create frame info
+            frames.append({
+                "id": frame_id,
+                "name": name,
+                "source": {"path": filename},
+                "line": line,
+                "column": 1,
+                "endLine": line,
+                "endColumn": 1
+            })
+            
+            # Cache frame for variable access
+            self.variables_cache[frame_id] = frame
+            
+            # MicroPython doesn't have f_back attribute
+            if hasattr(frame, 'f_back'):
+                frame = frame.f_back
+            else:
+                # Only return the current frame for MicroPython
+                break
+            frame_id += 1
+            
+        return frames
+        
+    def get_scopes(self, frame_id):
+        """Get variable scopes for a frame."""
+        scopes = [
+            {
+                "name": "Locals",
+                "variablesReference": frame_id * 1000 + 1,
+                "expensive": False
+            },
+            {
+                "name": "Globals", 
+                "variablesReference": frame_id * 1000 + 2,
+                "expensive": False
+            }
+        ]
+        return scopes
+        
+    def get_variables(self, variables_ref):
+        """Get variables for a scope."""
+        frame_id = variables_ref // 1000
+        scope_type = variables_ref % 1000
+        
+        if frame_id not in self.variables_cache:
+            return []
+            
+        frame = self.variables_cache[frame_id]
+        variables = []
+        
+        if scope_type == 1:  # Locals
+            var_dict = frame.f_locals if hasattr(frame, 'f_locals') else {}
+        elif scope_type == 2:  # Globals
+            var_dict = frame.f_globals if hasattr(frame, 'f_globals') else {}
+        else:
+            return []
+            
+        for name, value in var_dict.items():
+            # Skip private/internal variables
+            if name.startswith('__') and name.endswith('__'):
+                continue
+                
+            try:
+                value_str = str(value)
```

Comment:
I think this should be `value_str = repr(value)` . without that strings show without quotes etc.